### PR TITLE
Update ERC-5564: Move to Last Call

### PR DESCRIPTION
**Pull request by garyghayrat**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

**Review by nerolation — COMMENTED** [review]
LGTM

**Review by nerolation — APPROVED** [review]

File: ERCS/erc-5564.md [hunk 1]
```diff
@@ -4,7 +4,8 @@ title: Stealth Addresses
 description: Private, non-interactive transfers and interactions
```
Comment (SamWilsn) [inline ERCS/erc-5564.md:4]: What is a "non-interactive interaction"?

File: ERCS/erc-5564.md [hunk 2]
```diff
@@ -4,7 +4,8 @@ title: Stealth Addresses
...
 author: Toni Wahrstätter (@nerolation), Matt Solomon (@mds1), Ben DiFrancesco (@apbendi), Vitalik Buterin (@vbuterin)
 discussions-to: https://ethereum-magicians.org/t/eip-5566-stealth-addresses-for-smart-contract-wallets/10614
-status: Review
+status: Last Call
+last-call-deadline: 2024-03-31
```
Comment (SamWilsn) [inline ERCS/erc-5564.md:8]:
```suggestion
last-call-deadline: 2024-04-23
```

Giving some extra room for back-and-forth on this PR.

File: ERCS/erc-5564.md [hunk 3]
```diff
@@ -156,7 +157,7 @@ The new address format for the stealth meta-address extends the chain specific a
 Thus, a stealth meta-address on Ethereum has the following format:
 
 ```
-st:eth:0x<spendingKey><viewingKey>
+st:eth:0x<spendingPubKey><viewingPubKey>
 ``` 
 
 Stealth meta-addresses may be managed by the user and/or registered within a publicly available `Registry` contract, as delineated in [ERC-6538](./eip-6538). This provides users with a centralized location for identifying stealth meta-addresses associated with other individuals while simultaneously enabling recipients to express their openness to engage via stealth addresses.
```
Comment (SamWilsn) [inline ERCS/erc-5564.md:163]: ERC-6538 is Stagnant, and mentioning it here prevents moving this proposal out of Draft. Please either re-activate that proposal, or remove the reference here.

File: ERCS/erc-5564.md [hunk 4]
```diff
@@ -4,7 +4,8 @@ title: Stealth Addresses
 description: Private, non-interactive transfers and interactions
 author: Toni Wahrstätter (@nerolation), Matt Solomon (@mds1), Ben DiFrancesco (@apbendi), Vitalik Buterin (@vbuterin)
 discussions-to: https://ethereum-magicians.org/t/eip-5566-stealth-addresses-for-smart-contract-wallets/10614
-status: Review
+status: Last Call
+last-call-deadline: 2024-03-31
 type: Standards Track
 category: ERC
 created: 2022-08-13
```
Comment (SamWilsn) [inline ERCS/erc-5564.md:11]:
```suggestion
created: 2022-08-13
requires: 6538
```

From a quick read, it looks like it is necessary to read 6538 to understand how to use this proposal.

**Review by SamWilsn — CHANGES_REQUESTED** [review]
A few comments I can't place directly on the line:

> Furthermore, the schemeId MUST be added to [this overview](../assets/eip-5564/scheme_ids.md)

Once this proposal goes to final, that file will no longer be editable, so this requirement doesn't make sense.

---

> This MUST be a singleton contract, with one instance per chain

You don't specify how the reader can find the singleton contract. One common approach is to use a universal deployer or "Nick's Method", but I am really not an expert on that kind of thing.

edit: ah, I see. I don't understand why this info is in the Backwards Compatibility section however.

> The recipient's address and the `viewTag` MUST be included in the announcement event

The "MUST" here creates a requirement, but this sentence is not in the Specification section. If you're explaining a rule that already exists in that section, use the lowercase "must". If you do intend to create a new requirement, please move this to the Specification section.

**Comment by garyghayrat** [comment]
> A few comments I can't place directly on the line:
> 
> > Furthermore, the schemeId MUST be added to [this overview](../assets/eip-5564/scheme_ids.md)
> 
> Once this proposal goes to final, that file will no longer be editable, so this requirement doesn't make sense.
> 

I've updated the ERC to only specify `schemeId` `1` in this ERC.
> > This MUST be a singleton contract, with one instance per chain
> 
> You don't specify how the reader can find the singleton contract. One common approach is to use a universal deployer or "Nick's Method", but I am really not an expert on that kind of thing.
> 
> edit: ah, I see. I don't understand why this info is in the Backwards Compatibility section however.
> 

The Singleton contract address is in the `Deployment Method` section, but I've added another mention of the address right after the mention of the singleton contract.

> > The recipient's address and the `viewTag` MUST be included in the announcement event
> 
> The "MUST" here creates a requirement, but this sentence is not in the Specification section. If you're explaining a rule that already exists in that section, use the lowercase "must". If you do intend to create a new requirement, please move this to the Specification section.

I've added the `viewTag` requirement in the `Specification` section and updated to use lowercase "must" in the aforementioned  sentence.

File: ERCS/erc-5564.md (hunk 1 above)
Comment (garyghayrat) [reply ERCS/erc-5564.md]: I've changed it to `engagements`. 

The `non-interactive` part is where one can transfer funds to a randomly generated stealth address based on the receiver's `stealthMetaAddress`, without the recipient having to take any action.

The additional `interactions` or now `engagements` can be thought of as other actions that can be done with the randomly generated stealth address beyond regular transfers. Think messaging, voting, etc.

File: ERCS/erc-5564.md (hunk 2 above)
Comment (garyghayrat) [reply ERCS/erc-5564.md]: Updated in [7e3e4e5](https://github.com/ethereum/ERCs/pull/322/commits/7e3e4e585861a32d59978d3a159f6773afc28992)

File: ERCS/erc-5564.md (hunk 3 above)
Comment (garyghayrat) [reply ERCS/erc-5564.md:162]: We have submitted a PR at https://github.com/ethereum/ERCs/pull/259 to advance ERC-6538 to the review stage, currently waiting for editors' review.

File: ERCS/erc-5564.md (hunk 4 above)
Comment (garyghayrat) [reply ERCS/erc-5564.md:11]: Added in [7e3e4e5](https://github.com/ethereum/ERCs/pull/322/commits/7e3e4e585861a32d59978d3a159f6773afc28992)

File: ERCS/erc-5564.md (hunk 3 above)
Comment (garyghayrat) [reply ERCS/erc-5564.md:162]: https://github.com/ethereum/ERCs/pull/259 has been merged in!

**Review by moshahin01 — APPROVED** [review]

**Review by moshahin01 — APPROVED** [review]

**Review by moshahin01 — APPROVED** [review]

**Review by moshahin01 — CHANGES_REQUESTED** [review]
Ok

**Pull request merged**